tools/update.php

```diff
@@ -0,0 +1,890 @@
+    if ($diff_major < 0) {
+        echo '[!] Your version of LORIS is ahead of the latest release. If you '
+            . 'are not a developer working on the bleeding edge of LORIS, then '
+            . 'something has gone very wrong. SQL patches will not be applied.'
+            . PHP_EOL;
+        return array();
+    } else if ($diff_major === 0) {
+        // If major version is equal, check for difference in minor releases.
+        if ($diff_minor < 0) {
+            /* Major versions equal, but installed minor version is ahead of the
+             * latest release.
+             */
+            echo '[!] Your version of LORIS is ahead of the latest release. If '
+                . 'you are not a developer working on the bleeding edge of '
+                . 'LORIS, then something has gone very wrong. SQL patches will '
+                . 'not be applied.'
+                . PHP_EOL;
+            return array();
+        }
+        if ($diff_minor > 0) {
+            echo "[**] Latest version $version_to is ahead of installed "
+                . "$version_from by $diff_minor MINOR release(s)."
+                . PHP_EOL;
+        } else {
+            // If major and minor differences are 0, then LORIS is up-to-date.
+            echo "[**] Major and minor versions are equal. No patches to "
+                . "apply."
+                . PHP_EOL;
+            return array();
+        }
+    } else {
+        echo "[**] Latest version $version_to is ahead of installed "
+            . "$version_from by $diff_major MAJOR release(s)"
+            . PHP_EOL;
+    }
+
+    /* Following a discussion on this pull request, it was decided that patching
+     * SQL files automatically when a LORIS is more than one version behind
+     * should not be supported. This is to minimize errors that could impact
+     * data integrity.
+     * https://github.com/aces/Loris/pull/3677#discussion_r202814374
+     */
+    if ($diff_major > 1) {
+        echo "[!] Updating database schema when more than one major release "
+            . "behind is not supported. Patches will not be applied."
+            . PHP_EOL;
+        return array();
+    }
+    /* Similarly, applying patches across multiple minor versions is potentially
+     * risky. The user must explicitly opt-in to doing so.
+     */
+    if ($diff_minor > 1) {
+        echo "[!] Updating database schema when more than one minor release "
+            . "behind is considered risky. It is recommended to review the "
+            . "release notes before continuing and to proceed with caution."
+            . PHP_EOL;
+        $answers       = [
+            'y',
+            'N',
+        ];
+        $defaultAnswer = 'N';
+        writeQuestion('Apply patches anyway?', $answers);
+        $answer = readAnswer($answers, $defaultAnswer);
+        if ($answer != 'y') {
+            return array();
+        }
+    }
+
+
+    // For every major version released between the version that is installed
```

Comment:
comment might need to be changed "for every" no longer applies as only 1 patch can be supported